classes/Language.php

```diff
@@ -650,6 +616,26 @@ public static function getIsoById($id_lang)
         return false;
     }
 
+    public static function getJsonLanguageDetails($locale)
+    {
+        if (self::$_cache_all_language_json === null) {
+            self::$_cache_all_language_json = array();
+            $allLanguages = file_get_contents(_PS_ROOT_DIR_.self::ALL_LANGUAGES_FILE);
+            $allLanguages = json_decode($allLanguages, true);
+
+            $jsonLastErrorCode = json_last_error();
+            if (JSON_ERROR_NONE !== $jsonLastErrorCode) {
+                throw new \Exception('The legacy to standard locales JSON could not be decoded', $jsonLastErrorCode);
```

Comment:
I'm not sure this is what you want to do here. You should keep `500` as Exception error code and add jsonLastErrorCode in the Exception message.